src/compiler/checker.ts

```diff
@@ -14400,12 +14432,126 @@ namespace ts {
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        function checkUnusedModulePrivates(node: ModuleDeclaration): void {
+            if (compilerOptions.noUnusedLocals && !isInAmbientContext(node)) {
+                for (const key in node.locals) {
+                    if (hasProperty(node.locals, key)) {
+                        const local = node.locals[key];
+                        if (!local.hasReference && !local.exportSymbol) {
+                            if ((local.valueDeclaration && local.valueDeclaration.kind)) {
+                                error(local.valueDeclaration, Diagnostics._0_is_declared_but_never_used, key);
+                            }
+                            else if (local.declarations &&
+                                (local.declarations[0].kind === SyntaxKind.InterfaceDeclaration || local.declarations[0].kind === SyntaxKind.ModuleDeclaration)) {
+                                error(local.declarations[0], Diagnostics._0_is_declared_but_never_used, key);
+                            }
+                            else if (local.declarations && local.declarations[0].kind === SyntaxKind.ImportEqualsDeclaration) {
+                                error(local.declarations[0].name, Diagnostics._0_is_declared_but_never_used, key);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        function checkUnusedPrivates(node: ClassDeclaration): void {
+            if (compilerOptions.noUnusedLocals && !isInAmbientContext(node)) {
+                if (node.members) {
+                    for (const member of node.members) {
+                        if (member.kind === SyntaxKind.MethodDeclaration || member.kind === SyntaxKind.PropertyDeclaration) {
+                            if (isPrivateClassElement(member) && !member.symbol.hasReference) {
+                                error(member, Diagnostics._0_is_declared_but_never_used, member.symbol.name);
+                            }
+                        }
+                    }
+                }
+
+                if (node.typeParameters) {
+                    for (const typeParameter of node.typeParameters) {
+                        if (!typeParameter.symbol.hasReference) {
+                            error(typeParameter, Diagnostics._0_is_declared_but_never_used, typeParameter.symbol.name);
+                        }
+                    }
+                }
+            }
+        }
+
+        function isPrivateClassElement(node: ClassElement): boolean {
+            return (node.flags & NodeFlags.Private) !== 0;
+        }
+
+        function checkUnusedImports(node: SourceFile) {
+            if (compilerOptions.noUnusedLocals && !isInAmbientContext(node)) {

```

Comment:
i am not sure i see the point here. just merge them togather, and get rid of all the checks on the kinds.